manipulation/src/functTest/java/org/jboss/gm/manipulation/MultiModuleFunctionalTest.java

```diff
@@ -40,53 +40,38 @@ public void ensureProperPomGenerated() throws IOException, URISyntaxException, X
         assertThat(buildResult.task(":" + "generatePomFileForMainPublication").getOutcome()).isEqualTo(TaskOutcome.SUCCESS);
 
         // TODO: fix version assertions when plugin is implemented
-        assertRootPom(multiModuleRoot);
-        assertSubproject1Pom(multiModuleRoot);
-        assertSubproject2Pom(multiModuleRoot);
+        assertRootPom(multiModuleRoot, alignment);
+        assertSubproject1Pom(multiModuleRoot, alignment);
+        assertSubproject2Pom(multiModuleRoot, alignment);
     }
 
-    private void assertRootPom(File multiModuleRoot) throws IOException, XmlPullParserException {
-        final Path generatedPomPath = multiModuleRoot.toPath().resolve("build/publications/main/pom-default.xml");
-        assertThat(generatedPomPath).isRegularFile();
-
-        final MavenXpp3Reader reader = new MavenXpp3Reader();
-        final Model model = reader.read(new FileReader(generatedPomPath.toFile()));
-        assertThat(model.getGroupId()).isEqualTo("org.acme");
-        assertThat(model.getArtifactId()).isEqualTo("root");
-        assertThat(model.getVersion()).isEqualTo("1.1.2");
+    private void assertRootPom(File multiModuleRoot, AlignmentModel alignment) throws IOException, XmlPullParserException {
+        TestUtils.getModelAndCheckGAV(multiModuleRoot, alignment, "build/publications/main/pom-default.xml");
     }
 
-    private void assertSubproject1Pom(File multiModuleRoot) throws IOException, XmlPullParserException {
-        final Path generatedPomPath = multiModuleRoot.toPath().resolve("subproject1/build/publications/main/pom-default.xml");
-        assertThat(generatedPomPath).isRegularFile();
-
-        final MavenXpp3Reader reader = new MavenXpp3Reader();
-        final Model model = reader.read(new FileReader(generatedPomPath.toFile()));
-        assertThat(model.getGroupId()).isEqualTo("org.acme");
-        assertThat(model.getArtifactId()).isEqualTo("subproject1");
-        assertThat(model.getVersion()).isEqualTo("1.1.2");
-        assertThat(model.getDependencies())
+    private void assertSubproject1Pom(File multiModuleRoot, AlignmentModel alignment) throws IOException,
+            XmlPullParserException {
+        final Pair<Model, AlignmentModel.Module> modelAndModule = TestUtils.getModelAndCheckGAV(multiModuleRoot, alignment,
+                "subproject1/build/publications/main/pom-default.xml");
+        final AlignmentModel.Module module = modelAndModule.getRight();
```

Comment:
Maybe we should do something like the following in order to test the generated version as well:

```
        final AlignmentModel.Module module = modelAndModule.getRight();
        assertThat(modelAndModule.getLeft()).satisfies(model -> {
            assertThat(model.getVersion()).isEqualTo(module.getNewVersion());
            assertThat(model.getDependencies())
                    .extracting("artifactId", "version")
                    .containsOnly(
                            TestUtils.getAlignedTuple(module, "hibernate-core"),
                            TestUtils.getAlignedTuple(module, "commons-lang3", "3.8.1"),
                            TestUtils.getAlignedTuple(module, "undertow-core"));
        });
```